autocertkit/utils.py

```diff
@@ -945,6 +946,22 @@ def is_vf_disabled(session):
     return sum == 0
 
 
+def get_vf_driver_info(session, host, vm_ref, device):
+    cmd = b"%s -i %s" % (ETHTOOL, device)
+    cmd = binascii.hexlify(cmd)
```

Comment:
Why do we need to do this?

Reply:
See validate_exists() in autocertkit, it will verify args and only allow specified chars:
```
ARGUMENT_PATTERN = re.compile(r'^[a-zA-Z0-9_:\.\-,]+$')
...
def validate_exists(args, key, default=None, required=True):
        ...
        elif not ARGUMENT_PATTERN.match(args[key]):
```
Some shell metacharacter, e.g. single quotes is very general in shell command, so I convert shell command to hex string contains just 0-9 and a-f at first, and in shell_run() convert back.
Finally we can pass and run any shell command.
```
def shell_run(session, args):
    """General interface execute shell command(s) on host and even guest VM"""
    log.debug("shell run plugin call")
    vm_ref = validate_exists(args, 'vm_ref', None, False)
    username = validate_exists(args, 'username', "", False)
    password = validate_exists(args, 'password', "", False)
    cmd = validate_exists(args, 'cmd')
    cmd = binascii.unhexlify(cmd)
```